### DISCUSSION TITLE
Task name vs template name

### DISCUSSION THREAD

**Discussion by ranjithshegde** [stevearc/overseer.nvim#32] [Ideas]
`require("overseer.task_list").get_by_name(name)` expects the name to be the name of the command created by the template.

Is there a way to have the tasks issued by a template to carry the same name as the template name?

I am thinking this workflow:
- List all created or previously run tasks with a particular name, same as `overseer.task_list.get_by_name`
- If the task exists, rerun that task, maybe using `OverseerQuickAction restart` 
- If it does not, run the template that creates and executes the task
- A function that wraps around the above passed to a single keybinding

Right now I have one keybind to run a template by name. Pressing the key creates a new task of the same, as expected. Would be simpler if I could automate the above so that it precludes having to open overseer sidebar, choosing task and rerunning it manually.

**Comment by dzfrias** [comment]
I created a component called `unique` for this purpose. The component will restart an old task if another task with the same attribute (defaulted to the same name) is about to start. It then removes the task that's about to start and cancels its execution. The old task will be restarted.

It should fit your workflow if I'm understanding it correctly. Let me know if this works as you want it to. If you run into any problems with the component, feel free to tell me, as I made this component pretty recently.

To answer your original question, the name of the task depends on the template. You can have it have the same name as the template that created it as long as you specify that in the template definition. The task name is in the `name` field of the table returned by the `builder` function.

```lua
{
  desc = 'Make this task restart when tasks with the same arbitrary attribute are run',
  editable = false,
  params = {
    by = {
      desc = 'The attribute to use to determine if two tasks are the same',
      type = 'string',
      optional = true,
      default = 'name',
    },
  },
  constructor = function(params)
    return {
      on_pre_start = function(_, task)
        local tasks = require('overseer.task_list').list_tasks()
        for _, t in ipairs(tasks) do
          if t[params.by] == task[params.by] and t ~= task then
            task:dispose(true)
            t:restart(true)
            return false
          end
        end
      end,
    }
  end,
}
```

**Reply by ranjithshegde** [reply]
Thank you for this! Just tried it. It works exactly as I wanted my workflow to be.
I remember there being some docs with good examples. This component should definitely be added there.

**Reply by stevearc** [reply]
Added in a9b64d4e2d82c8710c8fb6d29ea63c54a4151653

Thanks for the great idea and the implementation!
